receiver/splunkhecreceiver/config.go

```diff
@@ -19,20 +19,18 @@ const (
 
 // Config defines configuration for the Splunk HEC receiver.
 type Config struct {
-	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
-
-	splunk.AccessTokenPassthroughConfig `mapstructure:",squash"`
-
-	Ack `mapstructure:"ack"`
-
+	// HecToOtelAttrs creates a mapping from HEC metadata to attributes.
+	HecToOtelAttrs splunk.HecToOtelAttrs `mapstructure:"hec_metadata_to_otel_attrs"`
+	Ack            `mapstructure:"ack"`
 	// RawPath for raw data collection, default is '/services/collector/raw'
 	RawPath string `mapstructure:"raw_path"`
 	// Splitting defines the splitting strategy used by the receiver when ingesting raw events. Can be set to "line" or "none". Default is "line".
 	Splitting SplittingStrategy `mapstructure:"splitting"`
 	// HealthPath for health API, default is '/services/collector/health'
 	HealthPath string `mapstructure:"health_path"`
-	// HecToOtelAttrs creates a mapping from HEC metadata to attributes.
-	HecToOtelAttrs splunk.HecToOtelAttrs `mapstructure:"hec_metadata_to_otel_attrs"`
+	// squash ensures fields are correctly decoded in embedded struct
+	confighttp.ServerConfig             `mapstructure:",squash"`
+	splunk.AccessTokenPassthroughConfig `mapstructure:",squash"`
```

Comment:
I'm not sure I'm happy with this change, tbh. The config is only one instance per collector. So, in the best case, we might be able to save one byte of memory with reduced readability. I'm OK with this if all collector leads agree with enforcing the field alignment. Please let me know if that's the case, I'm not aware. 

Don't get me wrong, I agree this can be important on a hot path.

Reply:
OK! will do.